### PR TITLE
fix(ButtonGroup): Align component with styleguide

### PR DESCRIPTION
**Pull request by renatogalvones**
## Description
https://jirasoftware.catho.com.br/browse/QTM-71

Extra:
- When passing an `as` prop to Button, the component was not behaving like expected. Fixed.

## Setup
to view the components behavior, run `yarn storybook`

## Review guide
- [ ] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [ ] Unit tests (yarn test:components)
- [ ] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [ ] Code review

### Browsers review
- [ ] Layout review
  - [ ] Edge
  - [ ] IE 11
  - [ ] Firefox
  - [ ] Chrome
  - [ ] Safari
  - [ ] Mobile
- [ ] Ux review validation

### CODE REVIEW

File: components/RadioGroup/RadioGroup.jsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ const Group = styled(FieldGroup).attrs({
 
 const RadioGroup = ({
   type,
+  size,
```
Comment (tcelestino) [inline components/RadioGroup/RadioGroup.jsx]: should it not update `interface RadioGroupProps` with new prop?

File: components/RadioGroup/RadioButton.jsx [hunk 2]
```diff
@@ -95,6 +95,7 @@ class Radio extends React.Component {
       inline,
       skin,
       theme,
+      size,
```
Comment (tcelestino) [inline components/RadioGroup/RadioButton.jsx]: should it not update `interface RadioButtonProps` with new prop?

File: components/Checkbox/CheckboxGroup.jsx [hunk 3]
```diff
@@ -42,6 +42,7 @@ class CheckboxGroup extends React.Component {
             error: Boolean(error),
             onChange: this._onChange,
             inline,
+            size,
```
[the remark this answers is not in the document]
Comment (renatogalvones) [reply components/Checkbox/CheckboxGroup.jsx]: There's no relation with `functions` in this case.
The `onChange` func is the one that is not being covered.

I think we can create a technical debit to increase the global percentage on our coverage, not specifically with this story.

image

**Pull request merged**